### PR TITLE
[ElmSharp] Add new APIs for automation test

### PR DESCRIPTION
**Pull request by arosis78**
Signed-off-by: Jeonghyun Yun <jh0506.yun@samsung.com>

### Description of Change ###
Add new property and events for ElmSharp app UI automation test 

### API Changes ###
Added:
public string EvasObject.AutomationId { get; set; } // Property
public event EventHandler Elementary.AutomationIdSet // Event
public event EventHandler EvasObject.Shown // Event
public event EventHandler EvasObject.Hidden // Event

### CODE REVIEW

File: src/ElmSharp/ElmSharp/Elementary.cs [hunk 1]
```diff
@@ -50,6 +50,17 @@ public static class Elementary
     {
         private static readonly string _themeFilePath = "/usr/share/elm-sharp/elm-sharp-theme.edj";
```
[the remark this answers is not in the document]
Comment (arosis78) [reply src/ElmSharp/ElmSharp/Elementary.cs]: Done

File: src/ElmSharp/ElmSharp/Elementary.cs [hunk 2]
```diff
@@ -48,7 +48,18 @@ public enum FocusAutoScrollMode
...
     public static class Elementary
     {
-        private static readonly string _themeFilePath = "/usr/share/elm-sharp/elm-sharp-theme.edj";
+        private const string _themeFilePath = "/usr/share/elm-sharp/elm-sharp-theme.edj";
+
+        /// <summary>
+        /// AutomationIdSet will be triggered when the AutomationId is set to the object.
+        /// </summary>
+        /// <since_tizen> preview </since_tizen>
+        public static event EventHandler AutomationIdSet;
```
Comment (rookiejava) [inline src/ElmSharp/ElmSharp/Elementary.cs]: How do you think about providing `Realized` or `EvasObjectRealized` event instead?

File: src/ElmSharp/ElmSharp/EvasObject.cs [hunk 3]
```diff
@@ -419,6 +442,25 @@ public EvasCanvas EvasCanvas
...
+            get
+            {
+                return _automationId;
+            }
+            set
+            {
+                if (_automationId != null)
+                    throw new InvalidOperationException("AutomationId may only be set one time.");
+                _automationId = value;
+                Elementary.SendAutomationIdSet(this);
```
Comment (rookiejava) [inline src/ElmSharp/ElmSharp/EvasObject.cs]: If you agree with my suggestion, this line should be moved to nearby `OnRealized()`.

**Comment by arosis78** [comment]
Abandoned.

**Pull request closed without merging**